src/Datadog.Trace/TracingProcessManager.cs

```diff
@@ -18,27 +22,59 @@ internal class TracingProcessManager
                 Name = "datadog-trace-agent",
                 ProcessPathKey = ConfigurationKeys.TraceAgentPath,
                 ProcessArgumentsKey = ConfigurationKeys.TraceAgentArgs,
+                PreStartAction = () =>
+                {
+                    var port = FreeTcpPort();
+                    var portString = port.ToString();
```

Comment:
Wouldn't this be `null` if an exception happens in `FreeTcpPort`?

Reply:
Yeah, the comment right below says that's okay: // Null ref here is okay, as it will force a retry loop
It will enter a retry loop, happy to make it an intentional exception or something.

Reply:
I didn't see that comment when I wrote my question. I guess I'd rather you make it intentional.